### DISCUSSION TITLE
Can I connect my 2 clusters with the wireguard cabledriver

### DISCUSSION THREAD

**Discussion by pengbinbin1** [submariner-io/submariner#2109] [Q&A]
I have 2 k8s clusters.One is on the Aliyun and have a public ip,another is under the NAT on my priveate network without public IP. Can I connect my 2 clusters with the **wireguard** cabledriver? If so,what the public IP should be set on my private network k8s cluster.

**Comment by sridhargaddam** [comment]
Submariner can be used to connect clusters in onPrem with public-cloud clusters. Please ensure that your Broker cluster is reachable from all the other clusters. So in your case, you can install the Broker cluster on the Aliyun cluster and as long as onPrem is able to connect to the public-cloud it should be fine. 

>If so,what the public IP should be set on my private network k8s cluster.

Submariner auto-discovers the public-ip for your clusters by making API calls to some external servers. So you do not have to set any explicit public-ip on your clusters.

You can take a look at various topologies and the associated configuration here https://submariner.io/operations/nat-traversal/
